benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +42,132 @@ def get_data_model_from_task(task, param_dict):
                     f"Unsupported data_model '{data_model}' defined for task '{task}'"
                 )
     return None
+
+
+def _partition_input_files(input_files, task_options):
+    """
+    This function partitions the input files into witness and other files.
+    The distinction is based on the file name of the witness file, which
+    is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Tuple of witness files and other files
+    """
+    witness_files = []
+    other_files = []
+    for file in input_files:
+        if Path(file).name == task_options.options.get("witness"):
+            witness_files.append(file)
+        else:
+            other_files.append(file)
+    return witness_files, other_files
+
+
+def get_witness_input_files(task):
+    """
+    This function returns the witness input files from the task.
+    They are identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: List of witness files
+    """
+    witness_files, _ = _partition_input_files(
+        task.input_files_or_identifier, task.options
```

Comment:
This is kind of a hack: It works because it is highly unlikely that someone creates a task with no input files but a witness. It also contradicts the documentation and naming of `_partition_input_files`, which only refers to files.

But I think we can keep it as is, if we add a comment that says that we can just pass through the task identifier as if it were a file name because it will not match a witness name.

Reply:
Sounds good, just added the comment.